### ISSUE TITLE
Commit 47d67c7 broke Attean!

### ISSUE THREAD

**Issue by kasei** [kjetilk/URI-NamespaceMap#14]
Appending a hash to namespace IRIs that don't end in a hash/slash broke Attean SPARQL parsing. This is part of the SPARQL test suite:

```
# return labels of items for the first 2 orders

PREFIX : <http://www.example.org>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?L
WHERE {
 ?O :hasItem [ rdfs:label ?L ] .
 {
 SELECT DISTINCT ?O  
 WHERE { ?O a :Order }
 ORDER BY ?O
 LIMIT 2
 }
} ORDER BY ?L
```

Even though the default prefix looks suspicious without a trailing slash/hash, I think the spec is pretty clear that the IRI produced in the subquery should be http://www.example.orgOrder, not http://www.example.org#Order.

**Comment by kjetilk** [comment]
Uh-oh... It was a problem with XSD that prompted me to do this. I'm on vacation, but I'll try to see if I can get some time to look into it. We should probably discuss on IRC.

**Comment by kjetilk** [comment]
Just a quick assessment of the breakage:

XML::CommonNS (CC @perigrin ) has ```http://www.w3.org/2001/XMLSchema```, and since it is a namespace name, is seems quite appropriate from the XML perspective. However, it is also [quite clear](https://www.w3.org/TR/xmlschema-2/#built-in-datatypes) clear that the full URI of a datatype needs to add a hash, e.g. ```http://www.w3.org/2001/XMLSchema#int```. RDF::NS::Curated, derives most of its prefixes from [the RDFa core initial context](https://www.w3.org/2011/rdfa-context/rdfa-1.1), which lists XSD with a trailing hash. So, it is a gab here needed to be bridged. Of course, I could special-case it for XSD, but that too feels awkward. Any ideas?

**Comment by kasei** [comment]
I think the gap was created by whoever it was in the RDF community who decided to use the same namespace IRI but add a hash when using it with RDF. They created the special case, so I think it wouldn't be inappropriate for RDF::NS::Curated to continue the special-casing (since it is RDF-specific, not just XML-related like XML::CommonNS).

**Comment by perigrin** [comment]
> On Jun 28, 2017, at 5:53 PM, Gregory Todd Williams <notifications@github.com> wrote:
> 
> I think the gap was created by whoever it was in the RDF community who decided to use the same namespace IRI but add a hash when using it with RDF. They created the special case, so I think it wouldn't be inappropriate for RDF::NS::Curated to continue the special-casing (since it is RDF-specific, not just XML-related like XML::CommonNS).

I think I agree with Greg here. I'm happy to make whatever changes are necessary and do a release but it sounds like the underlying modeling is broken between RDF-land and generic XML-Land and the fix should probably exist in the packages that model that split.

-Chris

**Comment by kjetilk** [comment]
OK, sounds reasonable. 

Given that XML::CommonNS and RDF::NS::Curated are just prefix-URI pairs, they should remain as they are. Which is a good thing. I'll special-case it in URI::NamespaceMap to fix this bug when I get around to it.

**Comment by kjetilk** [comment]
Does the previous commit look reasonable?

**Comment by kasei** [comment]
Yeah, I think so.

**Comment by kjetilk** [comment]
OK, good! Released as 1.04. :-)
